**How many positive integers $\leq$ 1260 are relatively prime to 1260?**

_____

- **Recall 1**: if two numbers $a$ and $b$ are relatively prime, then there is no integer $c > 1$ such that $c$ is a factor of both $a$ and $b$

- **Recall 2**: each integer has a *prime decomposition* that expresses the integer as a product of its prime factors
    - E.g. the prime decomposition of $22$ is $2\cdot11$
        - $120 = 2^{3}\cdot3\cdot5$

- If we combine these two properties, we can conclude that if two numbers are relatively prime, it means that they don't have any of the same prime factors
    - E.g. 21 and 25 are relatively prime, and $21 = 3\cdot7$, $25 = 5^{2}$

- The prime decomposition of 1260 is equal to:

$$
1260 = 2^{2}\cdot3^{2}\cdot5\cdot7
$$

- Therefore, we want to count the numbers less than 1260 that are not divisible by 2, 3, 5, and 7
    - This requires the same type of solution as the previous question
        - We won't calculate these by hand

In [4]:
from itertools import combinations
import numpy as np

In [5]:
list_primes = [2, 3, 5, 7]

In [22]:
total = 0

for i in range(1, 5):
    for comb in combinations(list_primes, i):
        list_choices = list(comb)
        product = np.prod(list_choices)
        val = (1260) // product
        total += (-1)**(i-1)*val
        print(list_choices, val)
print('total={}'.format(total))

[2] 630
[3] 420
[5] 252
[7] 180
[2, 3] 210
[2, 5] 126
[2, 7] 90
[3, 5] 84
[3, 7] 60
[5, 7] 36
[2, 3, 5] 42
[2, 3, 7] 30
[2, 5, 7] 18
[3, 5, 7] 12
[2, 3, 5, 7] 6
total=972


$$
\implies (\text{Number of positive integers  ≤  1260 are relatively prime to 1260}) = 630 + 420 + 252 + 180 - 210 - 126 - 90 - 84 - 60 - 36 + 42 + 30 + 18 + 12 - 6 = 972
$$